tests/psa-client-server/psasim/include/client.h

```diff
@@ -8,68 +6,42 @@
 #ifndef __PSA_CLIENT_H__
 #define __PSA_CLIENT_H__
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <stddef.h>
 
 #include "psa/crypto.h"
 
 #include "error_ext.h"
-/*********************** PSA Client Macros and Types *************************/
-
-#define PSA_FRAMEWORK_VERSION  (0x0100)
-
-#define PSA_VERSION_NONE       (0)
 
-/* PSA response types */
-#define PSA_CONNECTION_REFUSED PSA_ERROR_CONNECTION_REFUSED
```

Comment:
I would add a small block comment explaining the [summary and reasoning](https://github.com/Mbed-TLS/mbedtls/pull/9371/files#r1709525447) of the discussion on this issue.

There is definately work that needs to be done if PSA-SIM is picked up by other projects, so anything that could point contributors the right direction is welcome.

Reply:
I updated the `README` file in the `psasim` folder with the current status. I thought that would be a better place to talk about `psasim` overall structure.